strawberry/types/federation/object_type.py

```diff
@@ -0,0 +1,79 @@
+from typing import List, Optional, Union
+
+from strawberry.types import StrawberryObjectType, StrawberryUnion
+
+_OBJECT_FED_TYPE = Union[StrawberryObjectType, 'StrawberryObjectTypeFederation']
+
+
+# TODO: Should federation use composition or inheritance?
+#       i.e. should StrawberryObjectTypeFederation inherit from
+#       StrawberryObjectType or just hold a reference to one?
```

Comment:
federation is directives added to fields and objects, so I’d say we don’t need to extend a type, but we need a way to add directives/metadata to the field :)

Reply:
Could we just add a `directives` field, to which federation would add to?

Reply:
potentially! might be a clean approach :)